vanilla/src/launch/java/org/spongepowered/vanilla/launch/plugin/VanillaPluginManager.java

```diff
@@ -86,7 +88,9 @@ public void loadPlugins(final VanillaPluginPlatform platform) {
             } catch (final InstantiationException | IllegalAccessException | ClassNotFoundException | NoSuchMethodException | InvocationTargetException e) {
                 throw new RuntimeException(e);
             }
-            for (final PluginCandidate<PluginResource> candidate : candidates) {
+            final ResolutionResult<PluginResource> resolutionResult = DependencyResolver.resolveAndSortCandidates(candidates);
+            resolutionResult.printErrorsIfAny(platform.logger());
+            for (final PluginCandidate<PluginResource> candidate : resolutionResult.sortedSuccesses()) {
                 PluginContainer plugin = this.plugins.get(candidate.metadata().id());
```

Comment:
I wonder if we should be keeping around the failed plugins in the manager?

Reply:
The failed candidates are available in the result, so we can do that if you so wish, easily.

Reply:
I miss understood the code, perhaps.

Does `this.plugins` keep the failed plugins after the `DependencyResolver` runs?

Reply:
No, because the plugin containers are never constructed. I was just saying we can do that easily because I still have the failed candidates - we don't want to make the instances so we'd have to have logic for creating failed containers.

If we're going to do that, I think I'd want to make sure containers have a way of indicating their status.

Reply:
I was unclear with my original response, I didn't want them to stick around.

..that said I could see @Grinch wanting to show a plugin as *bad* on the GUI Plugins screen...it didn't load because of: <reason>. Granted it is hard for me to justify the additional backend work simply for that screen's benefit. I am not sure what, if anything else, would need to know what failed past your printout.

Reply:
Ah, right. For the absense of doubt:

The resolver returns all the candidates, but separated. `resolutionResult.sortedSuccesses()` will only proceed with those we know have a good dependency tree, they are the only ones that become `PluginContainer`s.

If you'd like to store failed plugins, the candidates are available through the other methods on the result.